router/sources/primary_name_tests.move

```diff
@@ -44,6 +44,37 @@ module router::primary_name_tests {
         }
     }
 
+    #[test(
+        router = @router,
+        aptos_names = @aptos_names,
+        aptos_names_v2 = @aptos_names_v2,
+        user1 = @0x077,
+        user2 = @0x266f,
+        aptos = @0x1,
+        foundation = @0xf01d
+    )]
+    fun test_set_primary_name_when_register(
+        router: &signer,
+        aptos_names: &signer,
+        aptos_names_v2: &signer,
+        user1: signer,
+        user2: signer,
+        aptos: signer,
+        foundation: signer
+    ) {
+        router::init_module_for_test(router);
+        let users = test_helper::e2e_test_setup(aptos_names, aptos_names_v2, user1, &aptos, user2, &foundation);
+        let user = vector::borrow(&users, 0);
+        let user_addr = address_of(user);
+        let domain_name = utf8(b"test");
+
+        router::register_domain(user, domain_name, SECONDS_PER_YEAR, option::none(), option::none());
+
+        let (primary_subdomain_name, primary_domain_name) = router::get_primary_name(user_addr);
+        assert!(*option::borrow(&primary_domain_name) == domain_name, 1);
+        assert!(option::is_none(&primary_subdomain_name), 2);
```

Comment:
can we also check it works after bumping mode to v1v2?

Reply:
added! the `set_primary_name_when_register()` is shared crossed modes. so I just repeat the test after I bumped the mode.